game-core/src/main/java/games/strategy/triplea/delegate/battle/BattleState.java

```diff
@@ -21,10 +29,12 @@
 
   Collection<Unit> getAttackingUnits();
 
-  Collection<Unit> getAttackingWaitingToDie();
-
   Collection<Unit> getDefendingUnits();
 
+  Collection<Unit> getUnits(EnumSet<Side> sides);
```

Comment:
nit, any consideration to make the API generic and allow for `EnumSet` to passed in as an implementation detail?
For example, if I had a `Set<Side>`, should I really be forced to convert it to an `EnumSet` to call this method? Is the runtime difference of forcing a collection copy enough that the `EnumSet` is that much more efficient?

Reply:
Perhaps even better, perhaps a vararg for this? In such a case a caller would not have to create a collection at all. WDYT?

Reply:
For yet more consideration, any thought to making this a default method?

It looks like these methods would give you the data access needed:
```

  @Override
  public Collection<Unit> getDefendingUnits() {
    return Collections.unmodifiableCollection(defendingUnits);
  }

  @Override
  public Collection<Unit> getRemainingAttackingUnits() {
    return Collections.unmodifiableCollection(attackingUnits);
  }
```

Reply:
Still more questions.. Should the above two methods be replaced by usages of this `getUnits(...)` method? Reducing the API surface of such a class would be a good thing, particularly removal of simple getter API methods.

Reply:
I hadn't thought of the vararg idea but I like it.  I'll make that change.

I've thought of replacing the `getAttackingUnits` and `getDefendingUnits` with just `getUnits(...)` but didn't want to make that change in this PR.  I've got one more step conversion PR in the process and then after that one, I'll do a clean up PR.  There's some tests that are now redundant that I can remove and the steps creation in `MustFightBattle#addFightSteps` will be ready for simplification.